dbms/src/Storages/KVStore/MultiRaft/Disagg/CheckpointIngestInfo.cpp

```diff
@@ -54,43 +55,59 @@ CheckpointIngestInfoPtr CheckpointIngestInfo::restore(
             peer_id,
             tmt.getKVStore()->getStoreID(std::memory_order_relaxed));
     }
-    ReadBufferFromMemory buf(page.data.begin(), page.data.size());
-    RUNTIME_CHECK_MSG(readBinary2<UInt8>(buf) == FAP_INGEST_INFO_PERSIST_FMT_VER, "wrong fap ingest info format");
 
-    std::vector<UInt64> restored_segments_id;
+    PS::V3::CheckpointProto::CheckpointIngestInfoPersisted ingest_info_persisted;
+    if (!ingest_info_persisted.ParseFromArray(page.data.data(), page.data.size()))
     {
-        auto count = readBinary2<UInt64>(buf);
-        for (size_t i = 0; i < count; ++i)
-        {
-            auto segment_id = readBinary2<UInt64>(buf);
-            restored_segments_id.push_back(segment_id);
-        }
-        remote_store_id = readBinary2<UInt64>(buf);
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "Can't parse CheckpointIngestInfo, region_id={} peer_id={} store_id={}",
+            region_id,
+            peer_id,
+            tmt.getKVStore()->getStoreID(std::memory_order_relaxed));
+    }
+
+    {
+        ReadBufferFromMemory buf(
+            ingest_info_persisted.region_info().data(),
+            ingest_info_persisted.region_info().size());
+        region = Region::deserialize(buf, proxy_helper);
     }
-    region = Region::deserialize(buf, proxy_helper);
 
     auto & storages = tmt.getStorages();
     auto keyspace_id = region->getKeyspaceID();
     auto table_id = region->getMappedTableID();
     auto storage = storages.get(keyspace_id, table_id);
-
-    auto log = DB::Logger::get("CheckpointIngestInfo");
     if (storage && storage->engineType() == TiDB::StorageEngine::DT)
     {
         auto dm_storage = std::dynamic_pointer_cast<StorageDeltaMerge>(storage);
         auto dm_context = dm_storage->getStore()->newDMContext(tmt.getContext(), tmt.getContext().getSettingsRef());
-        for (auto segment_id : restored_segments_id)
+        for (const auto & segment_pd : ingest_info_persisted.segments())
         {
-            restored_segments.emplace_back(DM::Segment::restoreSegment(log, *dm_context, segment_id));
+            ReadBufferFromString buf(segment_pd.segment_meta());
+            DM::Segment::SegmentMetaInfo segment_info;
+            readSegmentMetaInfo(buf, segment_info);
+
+            ReadBufferFromString buf_delta(segment_pd.delta_meta());
+            auto delta
+                = DM::DeltaValueSpace::restore(*dm_context, segment_info.range, buf_delta, segment_info.delta_id);
+            ReadBufferFromString buf_stable(segment_pd.stable_meta());
+            auto stable = DM::StableValueSpace::restore(*dm_context, buf_stable, segment_info.stable_id);
+
+            LOG_DEBUG(
+                log,
+                "Restore segments for checkpoint, remote_segment_id={} range={}",
+                segment_info.segment_id,
```

Comment:
maybe we should also print store id since segment id is meaningless for remote store.

Reply:
also fixed